pkg/build/strategies/sti/sti.go

```diff
@@ -345,7 +358,10 @@ func (builder *STI) Prepare(config *api.Config) error {
 
 	// fetch sources, for their .s2i/bin might contain s2i scripts
 	if len(config.Source) > 0 {
-		if builder.sourceInfo, err = builder.source.Download(config); err != nil {
+		startTime := time.Now()
+		builder.sourceInfo, err = builder.source.Download(config)
+		builder.result.BuildInfo.Steps = utilstatus.RecordStepInfo(builder.result.BuildInfo.Steps, api.StepDownloadSource, startTime)
```

Comment:
per my comment on your original version of this, when running under openshift, "Download" is really just a filesystem copy and isn't worth timing. (and shouldn't be called Download if we did decide to time it).